### PR TITLE
Validate: Warn if kube-proxy ipvs mode is detected

### PR DESCRIPTION
**Pull request by sridhargaddam**
Fixes issue: https://github.com/submariner-io/submariner-operator/issues/1150
Signed-Off-by: Sridhar Gaddam <sgaddam@redhat.com>

### CODE REVIEW

File: pkg/subctl/cmd/validate_kubeproxy.go [hunk 1]
```diff
@@ -0,0 +1,93 @@
...
+)
+
+var (
+	namespace string
+)
+
+var validateKubeProxyModeCmd = &cobra.Command{
+	Use:   "kube-proxy-mode",
+	Short: "Validate the Kube-proxy mode used in the cluster.",
+	Long:  "This command checks if kube-proxy mode is supported by Submariner.",
```
Comment (tpantelis) [inline pkg/subctl/cmd/validate_kubeproxy.go]:
```suggestion
	Short: "Validate the kube-proxy mode.",
	Long:  "This command checks if the kube-proxy mode is supported by Submariner.",
```

File: pkg/subctl/resource/schedule_pod.go [hunk 2]
```diff
@@ -0,0 +1,186 @@
...
+			},
+			Tolerations: []v1.Toleration{{Operator: v1.TolerationOpExists}},
+		},
+	}
+
+	pc := np.Config.ClientSet.CoreV1().Pods(np.Config.Namespace)
+	var err error
+	np.Pod, err = pc.Create(&networkPod)
+	if err != nil {
+		return errors.WithMessage(err, "error spawning the Pod")
```
Comment (tpantelis) [inline pkg/subctl/resource/schedule_pod.go]: The caller outputs a similar message.

```suggestion
		return err
```

File: pkg/subctl/resource/schedule_pod.go [hunk 3]
```diff
@@ -0,0 +1,186 @@
...
+	"fmt"
+
+	"github.com/pkg/errors"
+	"github.com/submariner-io/shipyard/test/e2e/framework"
+	v1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/client-go/kubernetes"
+)
+
+type NetworkPodType int
```
Comment (tpantelis) [inline pkg/subctl/resource/schedule_pod.go]: unused

File: pkg/subctl/resource/schedule_pod.go [hunk 4]
```diff
@@ -0,0 +1,186 @@
...
+	GatewayNode
+	NonGatewayNode
+)
+
+type NetworkingType bool
+
+const (
+	HostNetworking NetworkingType = true
+	PodNetworking  NetworkingType = false
+)
```
Comment (tpantelis) [inline pkg/subctl/resource/schedule_pod.go]: We can reuse these constants from shipyard framework.

File: pkg/subctl/resource/schedule_pod.go [hunk 5]
```diff
@@ -0,0 +1,186 @@
...
+	Command    string
+	Timeout    uint
+}
+
+type NetworkPod struct {
+	Pod       *v1.Pod
+	Config    *PodConfig
+	PodOutput string
+}
+
```
Comment (tpantelis) [inline pkg/subctl/resource/schedule_pod.go:43]: We should be able to reuse at least some of this from shipyard framework. I can look into that.

**Review by tpantelis — CHANGES_REQUESTED** [review]

File: pkg/subctl/resource/schedule_pod.go (hunk 4 above)
Comment (sridhargaddam) [reply pkg/subctl/resource/schedule_pod.go]: I thought about it, but my idea was to have validate_kubeproxy.go (package cmd) to depend ONLY on resource package and not depend on shipyard/framework.

But I can change it if you prefer to have it that way.

**Review by sridhargaddam — COMMENTED** [review]
Thanks for the review comments @tpantelis

File: pkg/subctl/resource/schedule_pod.go (hunk 4 above)
Comment (tpantelis) [reply pkg/subctl/resource/schedule_pod.go]: But you are using `framework.AwaitResultOrError` anyway 😄

Comment (sridhargaddam) [reply pkg/subctl/resource/schedule_pod.go]: haha.. no problem. I'll update validate_kubeproxy.go to use framework.XXXX directly

File: pkg/subctl/resource/schedule_pod.go (hunk 5 above)
Comment (sridhargaddam) [reply pkg/subctl/resource/schedule_pod.go:43]: I tried to reuse wherever possible, but if missed something please suggest. Thanks.

**Comment by sridhargaddam** [comment]
OVN job timed out during deployment stage itself.

**Review by tpantelis — APPROVED** [review]

File: pkg/subctl/resource/schedule_pod.go (hunk 5 above)
Comment (tpantelis) [reply pkg/subctl/resource/schedule_pod.go:43]: I created https://github.com/submariner-io/shipyard/issues/493

**Review by maayanf24 — APPROVED** [review]

**Pull request merged**